R/derive_vars_last_dose.R

```diff
@@ -137,6 +142,7 @@ derive_vars_last_dose <- function(dataset,
                                   single_dose_condition = EXDOSFRQ == "ONCE",
```

Comment:
@bundfussr are there full 1:1 equivalents of all these arguments into `derive_vars_joined()`? In particular, `dose_id` , `single_dose_condition` , and `traceability_vars` ? I'm not sure I can swap out everything for the: 

"In the initial release in which a function is deprecated the original function body must be replaced with a call to deprecate_warn() and **subsequently all arguments should be passed on to the new function.**"

Reply:
@zdz2101 , `dose_id` should be added to `order` and `traceability_vars` should be added to `new_vars`.

`single_dose_condition` does not have an equivalent in `derive_vars_joined()`. We could ignore it or perform the corresponding check before calling `derive_vars_joined()`.

Reply:
![image](https://github.com/pharmaverse/admiral/assets/43018762/49b22c15-fcad-4fd1-8cdb-19f78cafd321)

@bundfussr I think I'm getting a little confused how to incorporate `dose_date` argument into the `derive_vars_joined()` function, I believe what I have here in this screenshot does match the previous result but there are two caveats:

- the highlighted inclusion becomes necessary
- getting a duplicates warning/message

Reply:
@zdz2101 , I think `dose_date` should be added to `order`.

Why do you need to add `EXENDTM` to `new_vars`?

Reply:
The reason I added `EXENDTM` to `new_vars` because I think the filtering & mutate steps in the original function and joined function do not happen in the same order?

![image](https://github.com/pharmaverse/admiral/assets/43018762/fe52db4a-f048-43ac-bb23-994e294a384e)

Reply:
Otherwise I kept trying to play around adding the `dose_date` argument into `order` and `join_vars` but I seem to keep running into the line 341's `assert_date_frame` statement inside `derive_vars_joined()`

Reply:
```
  derive_vars_joined(dataset = dataset,
                     dataset_add = dataset_ex,
                     by_vars = by_vars,
                     order = expr_c(enexpr(dose_date), dose_id),
                     new_vars = expr_c({{new_vars}}, {{traceability_vars}}),
                     join_vars = expr_c(enexpr(dose_date), dose_id),
                     filter_add = {{filter_ex}},
                     filter_join = {{dose_date}} <= {{analysis_date}},
                     mode = "last",
                     )
```
seems to work. (I am not sure why your code does not work.)

I think that the change in the order of the `filter()` step does not matter because it is still _before_ the `select()` step. I.e., for filtering all variables of `dataset_add` are available. So `EXENDTM` does not need to be added to `new_vars`.

Reply:
yup that looked like that did the trick, just need to update tests for `derive_vars_last_dose()` as necessary now, but for curiosity @bundfussr would you be able to provide insight what the differences are between:

`enexpr(dose_date)`  vs `exprs({{dose_date}})` in this case?

Reply:
It seems that `enexpr(dose_date)` returns an expression and `exprs({{dose_date}})` returns a quosure.